src-java/server42/server42-control-storm-topology/src/main/java/org/openkilda/server42/control/topology/service/FlowRttService.java

```diff
@@ -81,13 +88,30 @@ public void activateFlowMonitoringForSwitch(SwitchId switchId) {
 
         flowByDirection.getOrDefault(true, Collections.emptyList())
                 .forEach(flow -> carrier.notifyActivateFlowMonitoring(flow.getFlowId(),
-                        switchId, flow.getSrcPort(), flow.getSrcVlan(), flow.getSrcInnerVlan(),
+                        switchId, flow.getDestSwitchId(), flow.getSrcPort(), flow.getSrcVlan(), flow.getSrcInnerVlan(),
                         true));
 
         flowByDirection.getOrDefault(false, Collections.emptyList())
                 .forEach(flow -> carrier.notifyActivateFlowMonitoring(flow.getFlowId(),
-                        switchId, flow.getDestPort(), flow.getDestVlan(), flow.getDestInnerVlan(),
+                        switchId, switchId, flow.getDestPort(), flow.getDestVlan(), flow.getDestInnerVlan(),
                         false));
+
+        activateHaFlowMonitoringForSwitch(switchId);
+    }
+
+    private void activateHaFlowMonitoringForSwitch(SwitchId switchId) {
+        haFlowRepository.findBySharedEndpointSwitchId(switchId)
+                .forEach(haFlow -> haFlow.getHaSubFlows().stream()
+                        .filter(haSubFlow -> !haSubFlow.isOneSwitchFlow())
```

Comment:
Maybe it makes sense to incorporate this condition into the repo, so that you don't fetch what you discard here.

Reply:
It isn't straightforward because the condition applied to the subflows inside the haFlow. Furthermore, I think it's better to have a more generic search that can be used from other places also the same filter is used in the classic flows so it's better to keep the consistency

Reply:
I think that if it is used in other places, it is an argument towards making it more efficient. As an option, you can move this code into the repository class and some other time improve its performance without changing anything in the service class.

Reply:
Is it used in other places but I meant the approach. I mean, in other repositories, it exists, for flows, haSubflows and haFlows. I'll take a look.